celery/app/task.py

```diff
@@ -525,6 +525,9 @@ def apply_async(self, args=None, kwargs=None, task_id=None, producer=None,
 
         preopts = self._get_exec_options()
         options = dict(preopts, **options) if options else preopts
+
+        options.setdefault('ignore_result', self.ignore_result)
```

Comment:
I think the only test cases missing now are the ones regarding passing correctly the `ignore_result` option.